app/src/ereferrals/java/org/eyeseetea/malariacare/strategies/DashboardActivityStrategy.java

```diff
@@ -227,12 +233,64 @@ private void openNewSurvey(Activity activity) {
         // Put new survey in session
         SurveyDB survey = new SurveyDB(null, programDB, Session.getUserDB());
         survey.save();
+        saveValuesFromIntent(activity, survey);
         Session.setMalariaSurveyDB(survey);
         //Look for coordinates
         prepareLocationListener(activity, survey);
         mDashboardActivity.initSurvey();
     }
 
+    private void saveValuesFromIntent(Activity activity, SurveyDB survey) {
+        if (activity.getIntent() != null && activity.getIntent().getExtras() != null){
+            String values = activity.getIntent().getExtras().getString(LoginActivityStrategy.VALUES_BUNDLE);
+            if(values !=null && !values.isEmpty()){
+                try {
+                    survey.save();
+                    parseJsonValuesAndSave(survey, values);
+                    if(survey.getValuesFromDB().size()>0) {
+                        DashboardActivity.dashboardActivity.setPreLoadSurveyOpenning(true);
+                    }
+                } catch (IOException e) {
+                    e.printStackTrace();
+                }
+            }
+            activity.getIntent().removeExtra(LoginActivityStrategy.VALUES_BUNDLE);
+            activity.getIntent().removeExtra(LoginActivityStrategy.CREATED_SURVEY_FROM_OTHER_APP);
+        }
+    }
+
+    private void parseJsonValuesAndSave(SurveyDB survey, String jsonValues) throws IOException {
```

Comment:
this method has more a one responsibility, can you, for example, create a mapper class that encapsulates mapping logic?